src/Expander.hs

```diff
@@ -115,12 +115,13 @@ expandModule thisMod src = do
   local (set (expanderLocal . expanderModuleName) thisMod) do
     lang <- mustBeModName (view moduleLanguage src)
     initializeLanguage lang
-    modBodyDest <- liftIO $ newModBodyPtr
-    modifyState $ set expanderModuleTop $ Just modBodyDest
+    declTreeDest <- liftIO $ newDeclTreePtr
+    vp <- newDeclValidityPtr
+    modifyState $ set expanderModuleTop $ Just declTreeDest
     decls <- addModuleScope (view moduleContents src)
-    forkExpandDecls modBodyDest decls
+    expandDeclForms declTreeDest mempty vp decls
```

Comment:
Here and in a few other places, I now call `expand***` instead of `forkExpand***`. An alternative would be to duplicate the `ExpandDeclForm` and `ExpandDeclForms` into two more tasks, which take a `ScopeSet` instead of an input `DeclValidityPtr`. Yet another alternative would be to create a new `DeclValidityPtr` and to immediately fill it with a `ScopeSet`, in this case `mempty`.

Reply:
If the whole thing works, then I see no reason to round-trip through the task system.